library/src/main/java/com/novoda/downloadmanager/lib/TarTruncator.java

```diff
@@ -18,11 +17,10 @@ public TarTruncator(DataWriter dataWriter) {
     public DownloadTask.State transferData(DownloadTask.State state, InputStream in) throws StopRequestException {
         DownloadTask.State newState = state;
         try {
-            byte[] buffer = new byte[BLOCK_SIZE];
-            int readLast = readBlock(in, buffer);
-            while (readsData(readLast) && isNotFullOfZeroes(buffer)) {
+            byte[] buffer = new byte[Constants.BUFFER_SIZE];
+            int readLast;
+            while ((readLast = readBuffer(in, buffer)) > 0 && isNotFullOfZeroes(buffer)) {
```

Comment:
Yep not nice but this is one of the rare cases where I allow myself to use it. The other options all have more trouble dealing with reading + copy of the bytes only when matching conditions.